autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +59,35 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
             updateFailures.put(autoscaleClusterId, failed);
             LOGGER.debug("Increased failed count '{}' for cluster '{}'", failed, cluster.getStackCrn());
+            // TODO: Determine what needs to be done if ForbiddenException occurs with endUser.
+            if (event.getCausedBy() instanceof ForbiddenException) {
```

Comment:
Also the check to say that a machineUser was being attempted - since that is where we potentially trigger a re-initalize.

Reply:
Updated with additional machineUser check